src/designer/elsa-workflows-studio/src/components/controls/elsa-monaco/elsa-monaco-utils.ts

```diff
@@ -8,30 +9,106 @@ export interface Monaco {
-  const baseUrl = libPath.startsWith('http') ? libPath : `${origin}/${libPath}`;
+      return win.monaco;
+    }
 
-  require.config({paths: {'vs': `${baseUrl}/vs`}});
-  win.MonacoEnvironment = {getWorkerUrl: () => proxy};
+    const origin = document.location.origin;
+    const baseUrl = libPath.startsWith('http') ? libPath : `${origin}/${libPath}`;
 
-  let proxy = URL.createObjectURL(new Blob([`
+    require.config({ paths: { 'vs': `${baseUrl}/vs` } });
+    win.MonacoEnvironment = { getWorkerUrl: () => proxy };
+
+    let proxy = URL.createObjectURL(new Blob([`
 	self.MonacoEnvironment = {
 		baseUrl: '${baseUrl}'
 	};
 	importScripts('${baseUrl}/vs/base/worker/workerMain.js');
-`], {type: 'text/javascript'}));
+`], { type: 'text/javascript' }));
 
-  return new Promise(resolve => {
-    require(["vs/editor/editor.main"], () => {
-      isInitialized = true;
-      resolve(win.monaco);
+    return new Promise(resolve => {
+      require(["vs/editor/editor.main"], () => {
+        isInitialized = true;
+        registerLiquid(win.monaco);
+        registerSQL(win.monaco);
+        resolve(win.monaco);
+      });
     });
   });
+
+
+}
+
+function registerLiquid(monaco: any) {
+  monaco.languages.register({ id: 'liquid' });
+
+  monaco.languages.registerCompletionItemProvider('liquid', {
+    provideCompletionItems: () => {
+      const autocompleteProviderItems = [];
+      const keywords = ['assign', 'capture', 'endcapture', 'increment', 'decrement',
+        'if', 'else', 'elsif', 'endif', 'for', 'endfor', 'break',
+        'continue', 'limit', 'offset', 'range', 'reversed', 'cols',
+        'case', 'endcase', 'when', 'block', 'endblock', 'true', 'false',
+        'in', 'unless', 'endunless', 'cycle', 'tablerow', 'endtablerow',
+        'contains', 'startswith', 'endswith', 'comment', 'endcomment',
+        'raw', 'endraw', 'editable', 'endentitylist', 'endentityview', 'endinclude',
+        'endmarker', 'entitylist', 'entityview', 'forloop', 'image', 'include',
+        'marker', 'outputcache', 'plugin', 'style', 'text', 'widget',
+        'abs', 'append', 'at_least', 'at_most', 'capitalize', 'ceil', 'compact',
+        'concat', 'date', 'default', 'divided_by', 'downcase', 'escape',
+        'escape_once', 'first', 'floor', 'join', 'last', 'lstrip', 'map',
+        'minus', 'modulo', 'newline_to_br', 'plus', 'prepend', 'remove',
+        'remove_first', 'replace', 'replace_first', 'reverse', 'round',
+        'rstrip', 'size', 'slice', 'sort', 'sort_natural', 'split', 'strip',
+        'strip_html', 'strip_newlines', 'times', 'truncate', 'truncatewords',
+        'uniq', 'upcase', 'url_decode', 'url_encode'];
+
+      for (let i = 0; i < keywords.length; i++) {
+        autocompleteProviderItems.push({ 'label': keywords[i], kind: monaco.languages.CompletionItemKind.Keyword });
+      }
+
+      return { suggestions: autocompleteProviderItems };
+    }
+  });
+}
+
+function registerSQL(monaco: any) {
```

Comment:
The term `SQL` should be renamed to `Sql` (in code only - not in comments etc.). The reason is that according to the naming convention we use, only two-letter acronyms (e.g. `IO`, `IP`) should be all-caps. Longer acronyms such as `Sql`, `Ado`, `Odbc` etc. should be pascal/camel cased (depending on the rules for classes, fields, locals etc.).

Reply:
Updated to `registerSql`, also, changed `editorVariables` to `EditorVariables`